### DISCUSSION TITLE
Create pull request with default merge method

### DISCUSSION THREAD

**Discussion by wemeetagain** [community/community#17435] [Pull Requests]
See https://docs.github.com/en/rest/pulls/pulls#create-a-pull-request

Automated PRs would benefit from being opened with a specific merge method set as default.
This would reduce human error in the process.

---

Add the following body parameter:

`default_merge_method` `string`

Default merge method to use. Possible values are `merge`, `squash` or `rebase`. Default is `merge`.

Can be one of: `merge`, `squash`, `rebase`

**Comment by dapplion** [comment]
This would be really cool, I want to enforce squash merges for a specific workflow while non-squash for another. Current default on last is a huge source of human error

**Comment by wemeetagain** [comment]
I just screwed up my git history again. Would really love to cut out the possibility of human error on this.

To reiterate our workflow:
- We have two long running branches, `unstable` and `stable`
- We **`squash`** all feature PRs to `unstable`
  - this is *critical* for us to generate coherent changelogs
- We occasionally create a release branch from `unstable`
- We then **`merge`** the release branch to `stable` and subsequently **`merge`** `stable` to `unstable`
  - this is *critical* for us to maintain the history of features between versions

And the issue:
- the default merge strategy logic is extremely prone to user error
  - our workflow involves `squash`ing 99% of the time, but `merge`ing at critical times
- merging the release branch into `stable` defaults to a `squash`, since prior PRs are usually feature PRs into `unstable`
  - this step requires extreme caution and is prone to user error
- merging `stable` to `unstable` may default to `squash` if a feature PR gets merged into `unstable`
  - this step requires extreme caution and is prone to user error
- the next subsequent feature PR will then default to `merge`!
  - this step requires extreme caution and is prone to user error

All in all, additional configuration of setting a default merge method for a PR would alleviate most or all of these issues.

**Reply by wemeetagain** [reply]
Related issue https://github.com/github-community/community/discussions/10809 discusses adding default merge method configurability at the branch level, rather than the PR level.

IMO both would be useful.
